Install versions

2bndy5 opened the pull request:
Modified the install script to allow specifying a version of the libs to install (defaults to master branch).

This was working about a month ago when I last tried it, but I haven't tried it since the v2.0 release crusade. The error output for installing a v2.0 mesh layer with a v1.x network layer may not be as intuitive as typical pkg managers would output because that would require a bit more comprehensive scripting.

2bndy5 committed:
ask for lib version when auto-installing

Still defaults to using cmake.
When cmake is not support in the specified version,
the installer falls
back to using the old makefiles (& configure script for RF24).

No version checking of RF24* deps are done;
that is too complex to implement in bash.

2bndy5 committed:
field-testing fixes

2bndy5 committed:
use absolute path instead of relative navigation

2bndy5 committed:
run ldconfig with sudo (for non-cmake builds)

2bndy5 committed:
add a newline before pyrf24 prompt

2bndy5 commented:
This works as intended. It would be nicer to have an integration with popular C++ pkg managers (like vcpkg or conan), but this will suffice for simplicity.

TMRh20 commented:
I think if the installer provides different versions, there could be an option at the very begining "Choose versions to install (Default: Install latest code from master) [y/N]?"

That way there are less options to choose from by default.
Other than that, I had to convert the file to UNIX format to run it, it seems to be in DOS format.

2bndy5 commented:
> I had to convert the file to UNIX format to run it, it seems to be in DOS format.

You mean convert CRLF to LF? I think I did that locally, but git on Windows needs to be told not to normalize the line endings on checkout. For other repos (like RF24), I use a .gitattributes file to do this without other team members having to manually `git config auto-crlf false`.

> "Choose versions to install (Default: Install latest code from master) [y/N]?"

I'm guessing you want a way to prevent all the version prompts for different libs. That does make sense. 👍🏼

2bndy5 committed:
use pre-emptive prompt; add gitattributes

2bndy5 commented:
I rebased to pull in changes to driver list prompt. Also added a preemptive prompt to use master branch by default.

I added a gitattributes file to tell `git checkout` to use LF on all *.sh files. Hopefully that should fix the Windows style line endings on checkout.

zaidalmurshi commented:
السلام عليكم ورحمة الله وبركاته
